cli/module_loader.rs

```diff
@@ -458,12 +469,50 @@ impl CliModuleLoader {
       return Err(generic_error("Attempted to load JSON module without specifying \"type\": \"json\" attribute in the import statement."));
     }
 
+    let code_cache = if module_type == ModuleType::JavaScript {
+      self.shared.code_cache.as_ref().and_then(|cache| {
+        let code_hash = self
+          .shared
+          .module_info_cache
+          .get_module_source_hash(specifier, code_source.media_type)
+          .ok()
+          .flatten();
+        let code_timestamp = specifier_to_file_path(specifier)
```

Comment:
Do we actually need the timestamp for code cache? Should the source hash be sufficient?

What would happen if we just avoided code cache for modules that didn't have a source cache in the `ModuleInfoCache`?

Reply:
> Do we actually need the timestamp for code cache? Should the source hash be sufficient?

2 reasons:
1. For some modules `ModuleInfoCache` doesn't provide the source hash
2. The other part of this PR deals with code cache for `op_eval_context` scripts, for which we don't compute/maintain hashes. Since `op_eval_context` scripts always come from the local file system, it seems that we could rely on timestamps rather than adding additional overhead to compute hashes.

> What would happen if we just avoided code cache for modules that didn't have a source cache in the ModuleInfoCache?

Yes, I think we could do that for ES modules. And then we can make it such that ES modules provide a hash, while `op_eval_context` scripts provide a timestamp.

Reply:
Imported npm modules don't have source hash in `ModuleInfoCache`. I'm assuming that's intentional and we don't want to introduce hash computation for them. I'm making a change to fallback to timestamps in that case.

Reply:
> The other part of this PR deals with code cache for op_eval_context scripts, for which we don't compute/maintain hashes. Since op_eval_context scripts always come from the local file system, it seems that we could rely on timestamps rather than adding additional overhead to compute hashes.

That's not entirely true, the other case is that when you `require()` something we use `op_eval_context` that has a generated source code (that wraps something loaded from the FS).

Reply:
For this case, what is the specifier?

Reply:
Ah in this particular case it will be `file:///<original specifier>` so I guess you're right here.